### PR TITLE
MM-33891 Add em dash support for commands

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
If an em dash is added to a command by introducing to dashes on iOS, it is considered as two dashes.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-33891

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: app/components/autocomplete/slash_suggestion/app_command_parser/app_command_parser.ts [hunk 1]
```diff
@@ -259,6 +259,12 @@ export class ParsedCommand {
                     this.i++;
                     break;
                 }
+                case '—': {
```
Comment (catalintomai) [inline app/components/autocomplete/slash_suggestion/app_command_parser/app_command_parser.ts:262]: wrt to "introduced when two '-' are set in iOS" ? - is it IOS only? (can they generate the em dash on other platforms as well?)

**Review by catalintomai — APPROVED** [review]

File: app/components/autocomplete/slash_suggestion/app_command_parser/app_command_parser.ts (hunk 1 above)
Comment (larkox) [reply app/components/autocomplete/slash_suggestion/app_command_parser/app_command_parser.ts:262]: The em dash can be created in any platform (worst case scenario, copy/paste). The problem with iOS is that by default, when you type to dashes, it automatically convert that to an em dash. Therefore why we have to deal with this case.

Comment (catalintomai) [reply app/components/autocomplete/slash_suggestion/app_command_parser/app_command_parser.ts:262]: But the code here is platform-agnostic, right?- meaning that if someone would ever type in the emdash directly we would address it as below (I understand that on iOS we get the emdash when the user types the "--" which is the more likely scenario to get the emdash).

**Review by levb — APPROVED** [review]

File: app/components/autocomplete/slash_suggestion/app_command_parser/app_command_parser.ts (hunk 1 above)
Comment (larkox) [reply app/components/autocomplete/slash_suggestion/app_command_parser/app_command_parser.ts:262]: Yes. The code is platform agnostic, and for simplicity, will be the same for webapp, so any version of the apps (Android, iOS, Webapp, Desktop...) will use an em dash as a double dash.

**Review by enahum — APPROVED** [review]

**Comment by DHaussermann** [comment]
@larkox this does work in that passing something in with an em dash such as `/servicenow config oauth —client_id pickles` 
will work as expected. 

However the solution does not provide that much value without the autocomplete also supporting `—`
When when I type the 2nd `-` and it becomes an em-dash I see `Parsing error Unable to identify argument` users would typically not proceed with this command. Is there another PR to accommodate autocomplete options treating `—` as `--`?

**Comment by larkox** [comment]
@DHaussermann No, this PR should have done exactly that. I will check what may have gone wrong there.

**Comment by DHaussermann** [comment]
@larkox Oh I see. In that case - here are my steps:
- Install Service Now
- Install this build on an iOS device (itms-services://?action=download-manifest&url=https://pr-builds.mattermost.com/mattermost-mobile/build-pr-5413-b5c0756/Mattermost_Beta.plist)
- Login from iOS
- Type `/servicenow config oauth --`
**Observed** parsing error

**Comment by larkox** [comment]
@DHaussermann fixed and ready to review

**Comment by DHaussermann** [comment]
@larkox  I can't build this PR. Seems like the build is failing. Can you please take a look?
https://app.circleci.com/pipelines/github/mattermost/mattermost-mobile/10370/workflows/d2dad958-eb4e-40cf-8c2b-9818b6aee9a2/jobs/16501

**Review by enahum — APPROVED** [review]

**Comment by enahum** [comment]
This build is failing as well as #5555

**Comment by larkox** [comment]
The build is fixed in this different PR: https://github.com/mattermost/mattermost-mobile/pull/5564
The code is exactly the same.

**Pull request closed without merging**